frontend/src/composables/useTransferNotifications.ts

```diff
@@ -0,0 +1,41 @@
+import type { Ref } from 'vue';
+import { computed, watch } from 'vue';
+import { useToast } from 'vue-toastification';
+
+import type { Transfer } from '@/actions/transfers';
+import TransferComplete from '@/components/notifications/TransferComplete.vue';
+
+export function useTransferNotifications(transfers: Ref<Array<Transfer>>) {
+  const toast = useToast();
+
+  const activeTransfers = computed(() => {
+    return transfers.value.filter((transfer) => !transfer.done);
+  });
+
+  const setupListeners = () => {
+    for (const transfer of activeTransfers.value) {
+      transfer.removeAllListeners();
+
+      transfer.on('completed', () => {
+        toast.success(
```

Comment:
I think we should also remove the listener here in order to clear the references. 

Also, in the case that a transfer fails, it might happen that a listener is never removed.

Reply:
Indeed, somehow overlooked this. Will implement accordingly.